airflow/models/dagcode.py

```diff
@@ -177,12 +177,13 @@ def get_code_by_fileloc(cls, fileloc: str) -> str:
         return cls.code(fileloc)
 
     @classmethod
-    def code(cls, fileloc) -> str:
+    @provide_session
+    def code(cls, fileloc, session: Session = NEW_SESSION) -> str:
         """Return source code for this DagCode object.
 
         :return: source code as string
         """
-        return cls._get_code_from_db(fileloc)
+        return cls._get_code_from_db(fileloc, session)
```

Comment:
This change is to avoid creating two different sessions because we use the method `code` and not the protected one.